lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js

```diff
@@ -20,12 +20,35 @@ import { ChartColors } from '../../Statistics/chartColors.js';
  * Render QC summary for given run and detector
  * @param {number} runNumber run number
  * @param {number} dplDetectorId id of the detector
+ * @param {object} production id of the production
+ * @param {number} production.dataPassId data pass id
+ * @param {number} production.simulationPassId simulation pass id
  * @param {QcSummary} qcSummary QC summary
  * @return {Component} component
  */
-const runDetectorAsyncQualityDisplay = (runNumber, dplDetectorId, qcSummary) => {
-    if (!qcSummary?.[runNumber]?.[dplDetectorId]) {
-        return h('.btn.white.bg-primary', 'QC');
+const runDetectorAsyncQualityDisplay = (runNumber, dplDetectorId, { dataPassId, simulationPassId }, qcSummary) => {
+    if (!qcSummary?.[runNumber]?.[dplDetectorId]) { // If there is no QC flag assigned
+        if (dataPassId) {
+            return frontLink(
```

Comment:
This bits of code, more specifically buttons to create a new flag duplicates a lot of code with the only difference being the class name `[...]data-pass` and `[...]simulation-pass` and the parameter ID being sent. 

Given that is also used in the same manner in other 2 file, please extract the type of button to a component and reuse it in all 4 places

Reply:
done